collector/lib/Utility.h

```diff
@@ -108,6 +108,36 @@ ScopedLock<Mutex> Lock(Mutex& mutex) {
 #define ssizeof(x) static_cast<ssize_t>(sizeof(x))
 
 std::optional<std::string_view> ExtractContainerIDFromCgroup(std::string_view cgroup);
+
+// A Holder refers to an object that it can own, or not.
```

Comment:
This sounds like a lifetime nightmare. Can't we do something similar to what protobuf does and simply use pointers? Or maybe make `SanitizeUTF8` return a `std::optional<std::string>`? That way if the return value is empty it would mean the original string is valid and, if it is not empty, it will hold the sanitized string.

Reply:
I see your point, I wasn't convinced myself.

The problem is with the immutability of std::string, which makes you copy and copy again. I will figure out a more maintainable way.